src/routes/api/playlists/user/+server.ts

```diff
@@ -0,0 +1,14 @@
+import { authenticate, getUserData, readPath } from '$lib/server/firebaseUtils';
+import { json } from '@sveltejs/kit';
+import type { RequestHandler } from './$types';
+
+export const GET = (async (event) => {
+    const { uid } = await authenticate(event);
+    const playlists = await readPath(`/playlists/user/${uid}`) || {};
+    const playlistArray = Object.entries(playlists).map(([key, val]) => val);
+    return json(playlistArray, { headers: [ ['Access-Control-Allow-Origin', "*"] ]});
```

Comment:
Your map is the same as `Object.values(playlists)` so this can be simplified
```suggestion
    return json(Object.values(playlists), { headers: [ ['Access-Control-Allow-Origin', "*"] ]});
```